doc/contributing.rst

```diff
@@ -0,0 +1,208 @@
+=============
+Contributing
+=============
+
+pyxem is meant to be a community maintained project. We welcome
+contributions in the form of bug reports, documentation, code, feature requests,
+and more. These guidelines provide resources on how best to contribute.
+
+For new users, checking out the `GitHub guides <https://guides.github.com>`_ are
+recommended.
+
+Questions, comments, and feedback
+=================================
+
+Have any questions, comments, suggestions for improvements, or any other
+inquiries regarding the project? Feel free to
+`ask a question <https://github.com/pyxem/pyxem/discussions>`_,
+`open an issue <https://github.com/pyxem/pyxem/issues>`_ or
+`make a pull request <https://github.com/pyxem/pyxem/pulls>`_ in our GitHub
+repository.
+
+Code of Conduct
+===============
+
+pyxem has a Code of conduct that should be honoured
+by everyone who participates in the pyxem community.
+
+.. _setting-up-a-development-installation:
+
+Setting up a development installation
+=====================================
+
+You need a `fork <https://guides.github.com/activities/forking/#fork>`_ of the
+`repository <https://github.com/pyxem/pyxem>`_ in order to make changes
+to pyxem.
+
+Make a local copy of your forked repository and change directories::
+
+    $ git clone https://github.com/your-username/pyxem.git
+    $ cd pyxem
+
+Set the ``upstream`` remote to the main pyxem repository::
+
+    $ git remote add upstream https://github.com/pyxem/pyxem.git
+
+We recommend installing in a `conda environment
+<https://conda.io/projects/conda/en/latest/user-guide/tasks/manage-environments.html>`_
+with the `Miniconda distribution
+<https://docs.conda.io/en/latest/miniconda.html>`_::
+
+   $ conda create --name pyxem
+   $ conda activate pyxem
+
+Then, install the required dependencies while making the development version
+available globally (in the ``conda`` environment)::
+
+   $ pip install --editable .[dev]
+
+This installs all necessary development dependencies, including those for
+running tests and building documentation.
+
+Code style
+==========
+
+The code making up pyxem is formatted closely following the `Style Guide for
+Python Code <https://www.python.org/dev/peps/pep-0008/>`_ with `The Black Code
+style <https://black.readthedocs.io/en/stable/the_black_code_style.html>`_. We
+use `pre-commit <https://pre-commit.com>`_ to run ``black`` automatically prior
+to each local commit. Please install it in your environment::
+
+    $ pre-commit install
+
+Next time you commit some code, your code will be formatted inplace according
+to our `black configuration
+<https://github.com/pyxem/pyxem/blob/master/pyproject.toml>`_.
+
+Note that ``black`` won't format `docstrings
+<https://www.python.org/dev/peps/pep-0257/>`_. We follow the `numpydoc
+<https://numpydoc.readthedocs.io/en/latest/format.html#docstring-standard>`_
+standard.
+
+Comment lines should preferably be limited to 72 characters.
```

Comment:
This hasn't been enforced before, I think, so I would remove it. Unless it is something that pyxem developers should start doing.